napi-inl.h

```diff
@@ -593,6 +593,10 @@ inline Symbol Symbol::New(napi_env env, napi_value description) {
   return Symbol(env, value);
 }
 
+inline Symbol Symbol::Public(napi_env env, const std::string& name) {
+  return Napi::Env(env).Global().Get("Symbol").As<Object>().Get(name).As<Symbol>();
```

Comment:
Is this right? It's getting a property on the global `Symbol` object with the given `name`, which by my reading of the JavaScript spec is undefined (unless `name` happens to be one of the defined static methods on `Symbol`). I'm not super experienced with symbols, but I would have expected this to invoke the `Symbol.for()` function, which creates a new public symbol.

Reply:
This is intentional. The idea is that instead of:

```Napi::Env(env).Global().Get("Symbol").As<Object>().Get("iterator").As<Symbol>();```

you can now do:

```Symbol::Public("iterator");```

The goal of this PR is to simplify adding symbol properties to classes; and one very common use case is defining `Symbol.iterator`, `Symbol.asyncIterator`.

I believe that the equivalent of `Symbol.for()` is `Symbol::New(...)`; so that API is already present.

Reply:
Oh, thanks. I see now, this is to reference what are commonly known as "well-known symbols".

Reference:
https://tc39.github.io/ecma262/#sec-well-known-symbols
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#Well-known_symbols

Based on some searching, the "public symbols" terminology appears to have been used some earlier in the development of the JS symbols concept, but now is not widely used anymore. So should this method be called `Symbol::WellKnown()` instead?

Reply:
Sure, that makes sense!